extensions/2.0/Vendor/EXT_mesh_features/README.md

```diff
@@ -616,13 +617,69 @@ Texture filtering must be `9728` (NEAREST), `9729` (LINEAR), or undefined, for a
+| ...                      | ...                      | ...             | ...                    |
+| `"UINT64"`               | `5121` <sup>3</sup>      | `"SINGLE"`      | `"SCALAR"`             |
+| ...                      | ...                      | ...             | ...                    |
+| `"INT8"`                 | `5120` (BYTE)            | `"SINGLE"`      | `"SCALAR"`             |
+| ...                      | ...                      | ...             | ...                    |
+| `"INT16"`                | `5122` (SHORT)           | `"SINGLE"`      | `"SCALAR"`             |
+| ...                      | ...                      | ...             | ...                    |
+| `"INT32"`                | `5124` (INT)<sup>4</sup> | `"SINGLE"`      | `"SCALAR"`             |
+| ...                      | ...                      | ...             | ...                    |
+| `"INT64"`                | `5121` <sup>3</sup>      | `"SINGLE"`      | `"SCALAR"`             |
+| ...                      | ...                      | ...             | ...                    |
+| `"FLOAT32"`              | `5126` (FLOAT)           | `"SINGLE"`      | `"SCALAR"`             |
+| ...                      | ...                      | ...             | ...                    |
+| `"FLOAT64"`              | `5121` <sup>3</sup>      | `"SINGLE"`      | `"SCALAR"`             |
+| ...                      | ...                      | ...             | ...                    |
+| `"BOOLEAN"`              | `5121` <sup>3</sup>      | `"SINGLE"`      | `"SCALAR"`             |
+| `"BOOLEAN"`              | `5121` <sup>3</sup>      | `"ARRAY"`       | `"SCALAR"`<sup>1</sup> |
+| `"STRING"`               | `5121` <sup>3</sup>      | `"SINGLE"`      | `"SCALAR"`             |
+| `"STRING"`               | `5121` <sup>3</sup>      | `"ARRAY"`       | `"SCALAR"`<sup>1</sup> |
+| `"ENUM"`                 | `5121` <sup>3</sup>      | `"SINGLE"`      | `"SCALAR"`             |
+| `"ENUM"`                 | `5121` <sup>3</sup>      | `"ARRAY"`       | `"SCALAR"`<sup>1</sup> |
+
+<small><sup>1</sup> Use "SCALAR" accessors for "ARRAY" property types, with `componentCount` for fixed-length arrays.</small><br>
+<small><sup>3</sup> `5125` (UNSIGNED_INT) disallowed by base glTF 2.0 specification except as `primitive.indices`. `EXT_mesh_features` overrides this restriction.</small><br>
+<small><sup>3</sup> `5121` (UNSIGNED_BYTE) used as general-purpose binary storage.</small><br>
+<small><sup>4</sup> `5124` (INT) disallowed by base glTF 2.0 specification. `EXT_mesh_features` overrides this restriction.</small>
+
+Each accessor `byteOffset` must be aligned to 4-byte boundaries, or to 8-byte boundaries for 64-bit property types.
+
+Buffer views used for property value accessors must be used only for that purpose — the same buffer view cannot also be used for vertex data, indices, or other values.
```

Comment:
I'm not sure if we want this restriction since one reason to move to accessor based storage is to allow property tables to point to the same accessors that glTF attributes point to.

One use case driving this is point clouds where a glTF may have vertex attributes for color, intensity, classification, etc that we want to reference in a property table. In this case `EXT_mesh_features` provides additional type information on the same view of the data.

Reply:
I see! If that's the case we'll probably need to say something about [`bufferView.target`](https://www.khronos.org/registry/glTF/specs/2.0/glTF-2.0.html#_bufferview_target) here. The base specification has requirements we are overriding:

> Buffer views used for images, vertex indices, vertex attributes, or inverse bind matrices MUST contain only one kind of data, i.e., the same buffer view MUST NOT be used both for vertex indices and vertex attributes.

I think this might be enough:

> Buffer views used for property value accessors may also contain vertex data, and the same accessor may itself be used both as a vertex attribute and a property value array. Such buffer views should specify `bufferView.target` with a value of _array buffer_. Buffer views containing property value accessors must not be used for other purposes like indices, inverse bind matrices, or animation samplers.